docs/DeveloperGuide.md

```diff
@@ -1,11 +1,267 @@
+The user can input as many sales as they like as long as the dish exists in the menu.
+
+Step 3. The user can generate the profit by inputting `profit`.
+
+
+#### 1.3.2 Design Considerations
+##### Aspect: Using a separate class to perform sale commands
+**Current Implementation**: Methods are stored in the sales class
+* Pros: Easy to understand and implement.
+* Cons: May make the sales class long and convoluted and may become difficult to find certain features or methods
+
+**Alternative**: Commands are stored in separate classes
+* Pros: Easier for a person who hasn't worked on the project to understand the structure
+* Cons: Takes longer to implement
 
-## Product Scope
+
+<a name="search-reservation"></a>
+
+### 1.4 [Proposed] Search reservation
+#### 1.4.1. Propose Implementation
+
+In the restaurant daily report, users can search against the reservation category by supplying either a reservation number or a date.
+
+The feature implements the following operations:
+* `SearchReservationCommand#parseInput()` - parse the user input for either reservation number or a date.
+* `SearchReservationCommand#execute()` - search along the reservation list for the target reservation(s).
+
+The following class diagram shows the structures relevant to the "search reservation" feature:
+<p align="center">
+    <img src="documentations\Sibing\ClassDiagramforSearchReservation.png">
+</p>
+
+Given below is an example usage scenario and how the search mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. An empty `reservation` will be initialized.
+
+Step 2. The user executes `add reservation; p/Peter; d/2020-03-12 12:00; n/3; c/98955555;` command to add a reservation into the `reservation` list.  
+Further, the user may add more reservations into the current `reservation` list.  
+Suppose the user executes `add reservation; p/Mary; d/2020-03-11 12:00; n/8; c/99998888;`  
+and `add reservation; p/Lisa; m/no spicy food please; d/2020-03-12 12:00; n/3; c/98889999;` as well.
+
+Step 3. The user can now search against the current `reservation` list to see if an reservation is stored in the `reservation` list.  
+If the user executes `search reservation; r/0;`, the following result will be displayed in the image.
+
+<p align="center">
+    <img src="documentations\Sibing\SearchByIndex.png">
+</p>
+
+If the user executes `search reservation; d/2020-03-12;`, the following result will be displayed in the image.
+
+<p align="center">
+    <img src="documentations\Sibing\SearchByDate.png">
+</p>
+
+If the user executes `search reservation; r/0; d/2020-03-13;`, the following result will be displayed in the image.
+
+<p align="center">
+    <img src="documentations\Sibing\SearchByIndexnDate.png">
+</p>
+
+The following sequence diagram shows the relevant interactions behind `search reservation`:
+
+<p align="center">
+    <img src="documentations\Sibing\SequenceDiagramforSearchReservation.png">
+</p>
```

Comment:
![complicateddiagram](https://user-images.githubusercontent.com/42376931/77989466-758de480-7351-11ea-8270-a2f446a9c779.JPG)
Perhaps you can consider putting some of the information in this diagram into other diagrams with the use of reference frames as it appears to be a bit overwhelming for the reader to digest. In particular, maybe the boxes pointed to by the red arrows can be put in separate diagrams using reference frames so that it reduces the nesting of the sequence diagram.

Reply:
Thanks for pointing out! I have added more `ref` frames into the diagram.